src/Features/Core/Portable/Intents/IIntentProvider.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Features.Intents
+{
+    internal interface IIntentProvider
```

Comment:
this is an internal API used to provide calculate the sln for applying a specific intent